### PR TITLE
Added file fetcher

### PR DESCRIPTION
**Pull request by aklyachkin**
Signed-off-by: aklyachkin <aklyachkin@gmail.com>

### Description

Added file fetcher to copy just a file (not the whole path).

--------------------------------------------------
/cc @chef/omnibus-maintainers <- This ensures the Omnibus Maintainers team are notified to review this PR.

#### Maintainers

Please ensure that you check for:

- [] If this change impacts git cache validity, it bumps the git cache
  serial number
- [] If this change impacts compatibility with omnibus-software, the
  corresponding change is reviewed and there is a release plan
- [] If this change impacts compatibility with the omnibus cookbook, the
  corresponding change is reviewed and there is a release plan

### CODE REVIEW

File: lib/omnibus/fetchers/file_fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    # @return [String]
+    #
+    def target_file
+      File.join(project_dir, File.basename(source_file))
+    end
+
+    #
+    # The shasum of the directory **inside** the project.
+    #
+    # @return [String]
```
Comment (srenatus) [inline lib/omnibus/fetchers/file_fetcher.rb]: [nit] This isn't correct, is it? I think I'd rather remove [line 119](https://github.com/chef/omnibus/pull/795/files#diff-33e4fde83dc6d509149df509f68bd6eeR119) and make this spec `@return [String, nil]` 🤔

File: lib/omnibus/fetchers/file_fetcher.rb [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    # The path on disk where the file is stored.
+    #
+    # @return [String]
+    #
+    def target_file
+      File.join(project_dir, File.basename(source_file))
+    end
+
+    #
+    # The shasum of the directory **inside** the project.
```
Comment (srenatus) [inline lib/omnibus/fetchers/file_fetcher.rb]: [nit] this still says "directory"

File: lib/omnibus/fetchers/file_fetcher.rb [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    # @return [String]
+    #
+    def target_shasum
+      @target_shasum ||= digest(target_file, :sha256)
+    rescue Errno::ENOENT
+      @target_shasum = 0
+    end
+
+    #
+    # The shasum of the directory **outside** of the project.
```
Comment (srenatus) [inline lib/omnibus/fetchers/file_fetcher.rb]: [nit] "directory"

**Comment by aklyachkin** [comment]
Thanks for the review!  I've copied path fetcher and changed it, but it seems, I forgot change some wordings :-) I did it know. And you're right it is nonsense to return 0 in the function, which should return "String". I changed it as you told.

**Pull request merged**